modules/opengl/src/shader/shader.cpp

```diff
@@ -118,9 +133,9 @@ Shader::Shader(Shader &&rhs) : id_{rhs.id_}, ready_(rhs.ready_), warningLevel_{r
 Shader &Shader::operator=(const Shader &that) {
     if (this != &that) {
         bool needRegister = false;
-        if (id_ == 0) {
+        if (!prog_.created()) {
             needRegister = true;
-            id_ = glCreateProgram();
+            prog_.id = glCreateProgram();
```

Comment:
Same here, this is leaking encapsulation...

Reply:
Same reasoning here, the ScopedShaderProgram is just responsible for scoping, not other parts since I could not find a good way of moving all responsibilities to it...

Reply:
That kind of shared responsibility does not look like a very attractive pattern...

Reply:
Agree, but I think it would require quite some changes otherwise. Possibly handing over a reference to the shader would possibly resolve it but that is also quite ugly. Happy to take suggestions on how to resolve it in a better way.

Reply:
Let me have a look on this...

Reply:
Easiest way would have been to just check for the vertex shader, before ever creating the program..